From 463fe1fbfa70064f9e0b3175f0f76e5877d84da6 Mon Sep 17 00:00:00 2001
From: "Yan, Zheng" <zheng.z.yan@intel.com>
Date: Tue, 21 Jan 2014 11:07:16 +0800
Subject: [PATCH 059/213] libceph: support CEPH_FEATURE_EXPORT_PEER
References: fate#318918
Git-commit: 80213a84a96c3040f5824bce646a184d5dd3dd2b
Patch-mainline: v3.14-rc1

Signed-off-by: Yan, Zheng <zheng.z.yan@intel.com>
Acked-by: David Disseldorp <ddiss@suse.de>

---
 include/linux/ceph/ceph_features.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/include/linux/ceph/ceph_features.h b/include/linux/ceph/ceph_features.h
index 5f42e44..d21b34d 100644
--- a/include/linux/ceph/ceph_features.h
+++ b/include/linux/ceph/ceph_features.h
@@ -80,7 +80,8 @@ static inline u64 ceph_sanitize_features(u64 features)
 	 CEPH_FEATURE_CRUSH_TUNABLES2 |		\
 	 CEPH_FEATURE_REPLY_CREATE_INODE |	\
 	 CEPH_FEATURE_OSDHASHPSPOOL |		\
-	 CEPH_FEATURE_CRUSH_V2)
+	 CEPH_FEATURE_CRUSH_V2 |		\
+	 CEPH_FEATURE_EXPORT_PEER)
 
 #define CEPH_FEATURES_REQUIRED_DEFAULT   \
 	(CEPH_FEATURE_NOSRCADDR |	 \
-- 
2.1.4

