From: Hariprasad S <hariprasad@chelsio.com>
Date: Wed, 24 Sep 2014 03:53:42 +0530
Subject: [PATCH 097/230] RDMA/cxgb4: Add missing neigh_release in find_route
Patch-mainline: v3.18-rc1
Git-commit: d480201b2261d26b4ca1fa4cc2bfc6cb644733a3
References: bsc#924381 FATE#318568 bsc#921338

Signed-off-by: Hariprasad Shenai <hariprasad@chelsio.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
Acked-by: David Chang <dchang@suse.com>
---
 drivers/infiniband/hw/cxgb4/cm.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/infiniband/hw/cxgb4/cm.c
+++ b/drivers/infiniband/hw/cxgb4/cm.c
@@ -417,6 +417,7 @@ static struct dst_entry *find_route(stru
 		return NULL;
 	if (!our_interface(dev, n->dev) &&
 	    !(n->dev->flags & IFF_LOOPBACK)) {
+		neigh_release(n);
 		dst_release(&rt->dst);
 		return NULL;
 	}
