From 97df93a988fb9361c2e159d4dccee1b45c2c5592 Mon Sep 17 00:00:00 2001
From: Ilya Dryomov <idryomov@gmail.com>
Date: Tue, 14 Apr 2015 20:55:13 +0300
Subject: [PATCH 208/213] crush: drop unnecessary include from mapper.c
References: fate#318918
Git-commit: 9be6df215a1baa57ccad0c388daeb2ad8f8d99ee
Patch-mainline: v4.1-rc1

Signed-off-by: Ilya Dryomov <idryomov@gmail.com>
Acked-by: David Disseldorp <ddiss@suse.de>

---
 net/ceph/crush/mapper.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/net/ceph/crush/mapper.c b/net/ceph/crush/mapper.c
index a1ef53c..5549fb6 100644
--- a/net/ceph/crush/mapper.c
+++ b/net/ceph/crush/mapper.c
@@ -20,7 +20,6 @@
 
 #include <linux/crush/crush.h>
 #include <linux/crush/hash.h>
-#include <linux/crush/mapper.h>
 
 /*
  * Implement the core CRUSH mapping algorithm.
-- 
2.1.4

