From: Hiral Shah <hishah@cisco.com>
Date: Thu, 20 Feb 2014 16:20:59 -0800
Subject: fnic: Incremented driver version
Patch-mainline: v3.13-rc1
References: bnc#863364,FATE#315811
Git-commit: 5ae303443094ab49fca41cf331d5af189995bead

Signed-off-by: Hiral Shah <hishah@cisco.com>
Acked-by: Hannes Reinecke <hare@suse.de>
---
 drivers/scsi/fnic/fnic.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/scsi/fnic/fnic.h b/drivers/scsi/fnic/fnic.h
index ce88951..1d3521e 100644
--- a/drivers/scsi/fnic/fnic.h
+++ b/drivers/scsi/fnic/fnic.h
@@ -39,7 +39,7 @@
 
 #define DRV_NAME		"fnic"
 #define DRV_DESCRIPTION		"Cisco FCoE HBA Driver"
-#define DRV_VERSION		"1.5.0.45"
+#define DRV_VERSION		"1.6.0.10"
 #define PFX			DRV_NAME ": "
 #define DFX                     DRV_NAME "%d: "
 
-- 
1.7.12.4

