From: Webb Scales <webb.scales@hp.com>
Date: Wed, 18 July 2014 00:00:00 -0000
Subject: hpsa: Revert some marketing names which are not yet ready for prime time
Git-commit: 3b7a45e5ba85dc79c7714edd9eee9aaed730cd6b (partial)
Patch-mainline: v3.16-rc1
References: bnc#887837


Signed-off-by: Shirish Pargaonkar <spargaonkar@suse.com>
---
diff --git a/drivers/scsi/hpsa.c b/drivers/scsi/hpsa.c
index c290551..765c884 100644
--- a/drivers/scsi/hpsa.c
+++ b/drivers/scsi/hpsa.c
@@ -188,9 +188,9 @@ static struct board_type products[] = {
 	{0x21C9103C, "Smart Array", &SA5_access},
 	{0x21CA103C, "Smart Array P246br", &SA5_access},
 	{0x21CB103C, "Smart Array P840", &SA5_access},
-	{0x21CC103C, "Smart Array P542t", &SA5_access},
-	{0x21CD103C, "Smart Array P240tr", &SA5_access},
-	{0x21CE103C, "Smart HBA H240tr", &SA5_access},
+	{0x21CC103C, "Smart Array", &SA5_access},
+	{0x21CD103C, "Smart Array", &SA5_access},
+	{0x21CE103C, "Smart Array", &SA5_access},
 	{0x00761590, "HP Storage P1224 Array Controller", &SA5_access},
 	{0x00871590, "HP Storage P1224e Array Controller", &SA5_access},
 	{0x007D1590, "HP Storage P1228 Array Controller", &SA5_access},
