From: Webb Scales <webbnh@hp.com>
Date: Thu, 25 Jun 2014 00:00:00 -0000
Subject: hpsa: update driver version to 3.4.4-126
Patch-mainline: Never, SLES specific
References: bnc#884422


Signed-off-by: Shirish Pargaonkar <spargaonkar@suse.com>
---
diff --git a/drivers/scsi/hpsa.c b/drivers/scsi/hpsa.c
index 972135d..46d9602 100644
--- a/drivers/scsi/hpsa.c
+++ b/drivers/scsi/hpsa.c
@@ -53,7 +53,7 @@
 #include "hpsa.h"
 
 /* HPSA_DRIVER_VERSION must be 3 byte values (0-255) separated by '.' */
-#define HPSA_DRIVER_VERSION "3.4.4-125"
+#define HPSA_DRIVER_VERSION "3.4.4-126"
 #define DRIVER_NAME "HP HPSA Driver (v " HPSA_DRIVER_VERSION ")"
 #define HPSA "hpsa"
 
