From: Len Brown <len.brown@intel.com>
Subject: intel_idle: support additional Broadwell model
References: fate#318535
Patch-Mainline: v4.0-rc1
Git-commit: bea57077e44ec9c1e6d3a3c142c8a3c0289e290d


Signed-off-by: Thomas Renninger <trenn@suse.de>

Signed-off-by: Len Brown <len.brown@intel.com>

diff --git a/drivers/idle/intel_idle.c b/drivers/idle/intel_idle.c
index 9cceacb..1bc0c170 100644
--- a/drivers/idle/intel_idle.c
+++ b/drivers/idle/intel_idle.c
@@ -727,6 +727,7 @@ static const struct x86_cpu_id intel_idle_ids[] = {
 	ICPU(0x46, idle_cpu_hsw),
 	ICPU(0x4d, idle_cpu_avn),
 	ICPU(0x3d, idle_cpu_bdw),
+	ICPU(0x47, idle_cpu_bdw),
 	ICPU(0x4f, idle_cpu_bdw),
 	ICPU(0x56, idle_cpu_bdw),
 	{}
