From: Dirk Brandewie <dirk.j.brandewie@intel.com>
Subject: intel_pstate: Add CPUID for BDW-H CPU
References: fate#318532
Patch-Mainline: v3.19-rc3
Git-commit: 43f8a966e91f387eabe85d2f2d12519c218f9dd0


Signed-off-by: Thomas Renninger <trenn@suse.de>

Add BDW-H to the list of supported processors.

Signed-off-by: Dirk Brandewie <dirk.j.brandewie@intel.com>
Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

diff --git a/drivers/cpufreq/intel_pstate.c b/drivers/cpufreq/intel_pstate.c
index ba35db0..ab2e100 100644
--- a/drivers/cpufreq/intel_pstate.c
+++ b/drivers/cpufreq/intel_pstate.c
@@ -793,6 +793,7 @@ static const struct x86_cpu_id intel_pstate_cpu_ids[] = {
 	ICPU(0x3f, core_params),
 	ICPU(0x45, core_params),
 	ICPU(0x46, core_params),
+	ICPU(0x47, core_params),
 	ICPU(0x4c, byt_params),
 	ICPU(0x4f, core_params),
 	ICPU(0x56, core_params),
