From: Kristen Carlson Accardi <kristen@linux.intel.com>
Subject: intel_pstate: Add support for SkyLake
References: fate#318533
Patch-Mainline: v4.0-rc1
Git-commit: 7ab0256e57ae4423fbfb6b6c1611578c634702c9


Signed-off-by: Thomas Renninger <trenn@suse.de>

Add SKL cpuid.

Signed-off-by: Kristen Carlson Accardi <kristen@linux.intel.com>
Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

diff --git a/drivers/cpufreq/intel_pstate.c b/drivers/cpufreq/intel_pstate.c
index 742eefb..7d9f822 100644
--- a/drivers/cpufreq/intel_pstate.c
+++ b/drivers/cpufreq/intel_pstate.c
@@ -825,6 +825,7 @@ static const struct x86_cpu_id intel_pstate_cpu_ids[] = {
 	ICPU(0x46, core_params),
 	ICPU(0x47, core_params),
 	ICPU(0x4c, byt_params),
+	ICPU(0x4e, core_params),
 	ICPU(0x4f, core_params),
 	ICPU(0x56, core_params),
 	{}
