From: Kristen Carlson Accardi <kristen@linux.intel.com>
Subject: intel_pstate: Add SKY-S support
References: bsc#949260
Patch-Mainline: v4.3-rc1
Git-commit: 1c9391238753ffb370f02743b5f43bac6dea304b


Signed-off-by: Thomas Renninger <trenn@suse.de>

Whitelist the SKL-S processor

Signed-off-by: Kristen Carlson Accardi <kristen@linux.intel.com>
Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

Index: linux-3.12-SLE12-SP1/drivers/cpufreq/intel_pstate.c
===================================================================
--- linux-3.12-SLE12-SP1.orig/drivers/cpufreq/intel_pstate.c	2015-10-08 16:07:52.620234022 +0200
+++ linux-3.12-SLE12-SP1/drivers/cpufreq/intel_pstate.c	2015-10-08 16:07:57.720518614 +0200
@@ -709,6 +709,7 @@ static const struct x86_cpu_id intel_pst
 	ICPU(0x4c, byt_params),
 	ICPU(0x4e, core_params),
 	ICPU(0x4f, core_params),
+	ICPU(0x5e, core_params),
 	ICPU(0x56, core_params),
 	{}
 };
