From: David Woodhouse <David.Woodhouse@intel.com>
Date: Wed, 19 Mar 2014 10:38:49 +0000
Subject: iommu/vt-d: Honour intel_iommu=sp_off for non-VMM domains
Git-commit: 214e39aa36c9c02355d388f20d83b93fc2fa7298
Patch-mainline: v3.15-rc2
References: fate#317112

Signed-off-by: David Woodhouse <David.Woodhouse@intel.com>
Acked-by: Borislav Petkov <bp@suse.de>
---
 drivers/iommu/intel-iommu.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/iommu/intel-iommu.c b/drivers/iommu/intel-iommu.c
index f5934fc2bbcc..c3d4bc91a189 100644
--- a/drivers/iommu/intel-iommu.c
+++ b/drivers/iommu/intel-iommu.c
@@ -1624,7 +1624,11 @@ static int domain_init(struct dmar_domain *domain, int guest_width)
 	else
 		domain->iommu_snooping = 0;
 
-	domain->iommu_superpage = fls(cap_super_page_val(iommu->cap));
+	if (intel_iommu_superpage)
+		domain->iommu_superpage = fls(cap_super_page_val(iommu->cap));
+	else
+		domain->iommu_superpage = 0;
+
 	domain->nid = iommu->node;
 
 	/* always allocate the top pgd */

