From: Saurav Kashyap <saurav.kashyap@qlogic.com>
Date: Wed, 30 Oct 2013 03:38:28 -0400
Subject: [SCSI] qla2xxx: Adding MAINTAINERS for qla2xxx FC-SCSI driver
References: bnc#853689,FATE#315933
Patch-mainline: v3.14-rc1
Git-commit: 2c804eb04945dd3e4f154168a06c8a6b4ce5a748

Signed-off-by: Giridhar Malavali <giridhar.malavali@qlogic.com>
Signed-off-by: Saurav Kashyap <saurav.kashyap@qlogic.com>
Signed-off-by: James Bottomley <JBottomley@Parallels.com>
Acked-by: Hannes Reinecke <hare@suse.de>
---
 MAINTAINERS | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/MAINTAINERS b/MAINTAINERS
index 1344816..0fbd08d 100644
--- a/MAINTAINERS
+++ b/MAINTAINERS
@@ -6901,8 +6901,7 @@ S:	Maintained
 F:	drivers/scsi/qla1280.[ch]
 
 QLOGIC QLA2XXX FC-SCSI DRIVER
-M:	Andrew Vasquez <andrew.vasquez@qlogic.com>
-M:	linux-driver@qlogic.com
+M:	qla2xxx-upstream@qlogic.com
 L:	linux-scsi@vger.kernel.org
 S:	Supported
 F:	Documentation/scsi/LICENSE.qla2xxx
-- 
1.7.12.4

