From: Jiri Pirko <jiri@resnulli.us>
Date: Thu, 23 Jan 2014 16:51:27 +0100
Subject: [PATCH 165/181] rtnetlink: add missing IFLA_BOND_AD_INFO_UNSPEC
Patch-mainline: v3.14-rc1
Git-commit: 237266f76d417b5fb1984ac3774e3a4fe070fe5d
References: fate#316924 bnc#875631 bnc#876145

Signed-off-by: Jiri Pirko <jiri@resnulli.us>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Michal Kubecek <mkubecek@suse.cz>

---
 include/uapi/linux/if_link.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/uapi/linux/if_link.h b/include/uapi/linux/if_link.h
index 1f5f3b8..ab3833a 100644
--- a/include/uapi/linux/if_link.h
+++ b/include/uapi/linux/if_link.h
@@ -361,6 +361,7 @@ enum {
 #define IFLA_BOND_MAX	(__IFLA_BOND_MAX - 1)
 
 enum {
+	IFLA_BOND_AD_INFO_UNSPEC,
 	IFLA_BOND_AD_INFO_AGGREGATOR,
 	IFLA_BOND_AD_INFO_NUM_PORTS,
 	IFLA_BOND_AD_INFO_ACTOR_KEY,
-- 
1.8.1.4

