From: Hannes Reinecke <hare@suse.de>
Date: Thu, 6 Mar 2014 13:26:09 +0100
Subject: dm: Initialize flush_rq
References: bnc#866989
Patch-Mainline: Not yet

flush_rq wasn't initialized, so do it now.

Signed-off-by: Hannes Reinecke <hare@suse.de>

diff --git a/drivers/md/dm.c b/drivers/md/dm.c
index f404885..bd615a1 100644
--- a/drivers/md/dm.c
+++ b/drivers/md/dm.c
@@ -2363,10 +2363,17 @@ static int dm_init_request_based_queue(struct mapped_device *md)
 	if (md->queue->elevator)
 		return 1;
 
+	md->queue->flush_rq = kzalloc(sizeof(struct request), GFP_KERNEL);
+	if (!md->queue->flush_rq)
+		return 0;
+
 	/* Fully initialize the queue */
 	q = blk_init_allocated_queue(md->queue, dm_request_fn, NULL);
-	if (!q)
+	if (!q) {
+		kfree(md->queue->flush_rq);
+		md->queue->flush_rq = NULL;
 		return 0;
+	}
 
 	md->queue = q;
 	dm_init_md_queue(md);
