From: Olaf Hering <olaf@aepfle.de>
Date: Wed, 19 Feb 2014 18:19:31 +0100
Subject: drm/cirrus: bind also to qemu-xen-traditional
Patch-mainline: submitted (lkml 2014-04-11)
References: bnc#864657, bnc#864288

Signed-off-by: Olaf Hering <olaf@aepfle.de>
Acked-by: <ohering@suse.de>
---
 drivers/gpu/drm/cirrus/cirrus_drv.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/gpu/drm/cirrus/cirrus_drv.c b/drivers/gpu/drm/cirrus/cirrus_drv.c
index 138364d..fe7c1bf 100644
--- a/drivers/gpu/drm/cirrus/cirrus_drv.c
+++ b/drivers/gpu/drm/cirrus/cirrus_drv.c
@@ -31,6 +31,8 @@ static struct drm_driver driver;
 static DEFINE_PCI_DEVICE_TABLE(pciidlist) = {
 	{ PCI_VENDOR_ID_CIRRUS, PCI_DEVICE_ID_CIRRUS_5446, 0x1af4, 0x1100, 0,
 	  0, 0 },
+	{ PCI_VENDOR_ID_CIRRUS, PCI_DEVICE_ID_CIRRUS_5446, PCI_VENDOR_ID_XEN, 0x0001, 0,
+	  0, 0 },
 	{0,}
 };
 
