From: Trond Myklebust <Trond.Myklebust@netapp.com>
Date: Thu, 26 Sep 2013 10:18:04 -0400
Subject: SUNRPC: Only update the TCP connect cookie on a successful connect
Git-commit: 8b71798c0d389d4cadc884fc7d68c61ee8cd4f45
Patch-mainline: v3.13
References: bnc#868923

Signed-off-by: Trond Myklebust <Trond.Myklebust@netapp.com>
Acked-by: NeilBrown <neilb@suse.de>

---
 net/sunrpc/xprtsock.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- linux-3.12-SLE12.orig/net/sunrpc/xprtsock.c
+++ linux-3.12-SLE12/net/sunrpc/xprtsock.c
@@ -1527,6 +1527,7 @@ static void xs_tcp_state_change(struct s
 			transport->tcp_copied = 0;
 			transport->tcp_flags =
 				TCP_RCV_COPY_FRAGHDR | TCP_RCV_COPY_XID;
+			xprt->connect_cookie++;
 
 			xprt_wake_pending_tasks(xprt, -EAGAIN);
 		}
@@ -2184,7 +2185,6 @@ static int xs_tcp_finish_connecting(stru
 	case 0:
 	case -EINPROGRESS:
 		/* SYN_SENT! */
-		xprt->connect_cookie++;
 		if (xprt->reestablish_timeout < XS_TCP_INIT_REEST_TO)
 			xprt->reestablish_timeout = XS_TCP_INIT_REEST_TO;
 	}
