From: Daniel Baluta <dbaluta@ixiacom.com>
Date: Fri, 31 Jan 2014 09:50:12 +0200
Subject: vxlan: remove extra newline after function definition
Patch-mainline: v3.14-rc3
Git-commit: 4fe46b9a4d0b5eef96867e6d5134159e5a65d2a5
References: bsc#938963 FATE#319084

Signed-off-by: Daniel Baluta <dbaluta@ixiacom.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Benjamin Poirier <bpoirier@suse.com>
---
 drivers/net/vxlan.c |    1 -
 1 file changed, 1 deletion(-)

--- a/drivers/net/vxlan.c
+++ b/drivers/net/vxlan.c
@@ -472,7 +472,6 @@ static inline struct hlist_head *vxlan_f
 /* Look up Ethernet address in forwarding table */
 static struct vxlan_fdb *__vxlan_find_mac(struct vxlan_dev *vxlan,
 					const u8 *mac)
-
 {
 	struct hlist_head *head = vxlan_fdb_head(vxlan, mac);
 	struct vxlan_fdb *f;
