From 8cc7212a036118fcb5cfbbdb013c5032677bbd23 Mon Sep 17 00:00:00 2001
From: Vladimir Davydov <vdavydov@parallels.com>
Date: Thu, 30 Jan 2014 19:57:54 +0400
Subject: idr: remove unused prototype of idr_free()
References: fate#318475
Patch-mainline: v3.15
Git-commit: 8cc7212a036118fcb5cfbbdb013c5032677bbd23

There is no such function. Remove the redundant prototype.

Signed-off-by: Vladimir Davydov <vdavydov@parallels.com>
Acked-by: Tejun Heo <tj@kernel.org>
Signed-off-by: Jiri Kosina <jkosina@suse.cz>
Acked-by: Dinar Valeev <dvaleev@suse.com>

diff --git a/include/linux/idr.h b/include/linux/idr.h
index 871a213..ff94a48 100644
--- a/include/linux/idr.h
+++ b/include/linux/idr.h
@@ -82,7 +82,6 @@ int idr_for_each(struct idr *idp,
 void *idr_get_next(struct idr *idp, int *nextid);
 void *idr_replace(struct idr *idp, void *ptr, int id);
 void idr_remove(struct idr *idp, int id);
-void idr_free(struct idr *idp, int id);
 void idr_destroy(struct idr *idp);
 void idr_init(struct idr *idp);
 
-- 
cgit v0.10.2

