From ac5d79cf0367e906e50a77585892475cafa9a49a Mon Sep 17 00:00:00 2001
From: Oliver Neukum <oneukum@suse.com>
Date: Wed, 16 Mar 2016 11:30:22 +0100
Subject: [PATCH] wacom: fix crash in probe()
Patch-Mainline: Never (upstream has been rewritten and is not vulnerable)
References: bnc#970909, CVE-2016-3139

To guard agaionst malicious devices a sanity check is needed.

Signed-off-by: Oliver Neukum <oneukum@suse.de>
---
 drivers/input/tablet/wacom_sys.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/input/tablet/wacom_sys.c b/drivers/input/tablet/wacom_sys.c
index 44a1fb6..d318299 100644
--- a/drivers/input/tablet/wacom_sys.c
+++ b/drivers/input/tablet/wacom_sys.c
@@ -1297,6 +1297,10 @@ static int wacom_probe(struct usb_interface *intf, const struct usb_device_id *i
 	strlcat(wacom->phys, "/input0", sizeof(wacom->phys));
 
 	endpoint = &intf->cur_altsetting->endpoint[0].desc;
+	if (!endpoint) {
+		error = -ENODEV;
+		goto fail3;
+	}
 
 	/* set the default size in case we do not get them from hid */
 	wacom_set_default_phy(features);
-- 
2.1.4

