From: Goldwyn Rodrigues <rgoldwyn@suse.com>
Date: Fri, 7 Mar 2014 10:30:43 -0600
Subject: [PATCH 02/57] Add number of nodes to bitmap structure for clustering
Git-commit: 183bdf5106af069a774146c4f910a80ad1f57485
Patch-mainline: v4.1
References: fate#316335

Signed-off-by: Goldwyn Rodrigues <rgoldwyn@suse.com>
Acked-by: NeilBrown <neilb@suse.com>

---
 drivers/md/bitmap.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/md/bitmap.h b/drivers/md/bitmap.h
index df4aeb6ac6f0..c0462f67a243 100644
--- a/drivers/md/bitmap.h
+++ b/drivers/md/bitmap.h
@@ -131,7 +131,8 @@ typedef struct bitmap_super_s {
 	__le32 sectors_reserved; /* 64 number of 512-byte sectors that are
 				  * reserved for the bitmap. */
 
-	__u8  pad[256 - 68]; /* set to zero */
+	__le32 nodes;        /* 68 the maximum number of nodes in cluster. */
+	__u8  pad[256 - 72]; /* set to zero */
 } bitmap_super_t;
 
 /* notes:
-- 
2.5.0

