From: Tony Jones <tonyj@suse.de>
Subject: Add external support and no support module to bad taint 

Patch-mainline: Never, SLES feature
References: bnc#870450
Target: SLE-12

Acked-by: Lee, Chun-Yi <jlee@suse.com>

Index: linux-3.12-SLE12/kernel/tracepoint.c
===================================================================
--- linux-3.12-SLE12.orig/kernel/tracepoint.c
+++ linux-3.12-SLE12/kernel/tracepoint.c
@@ -634,7 +634,11 @@ EXPORT_SYMBOL_GPL(tracepoint_iter_reset)
 bool trace_module_has_bad_taint(struct module *mod)
 {
 	return mod->taints & ~((1 << TAINT_OOT_MODULE) | (1 << TAINT_CRAP) |
-			       (1 << TAINT_UNSIGNED_MODULE));
+			       (1 << TAINT_UNSIGNED_MODULE)
+#ifdef CONFIG_SUSE_KERNEL_SUPPORTED
+			       | (1 << TAINT_EXTERNAL_SUPPORT) | (1 << TAINT_NO_SUPPORT)
+#endif
+			       );
 }
 
 static int tracepoint_module_coming(struct module *mod)
