From: Stephen Rothwell <sfr@canb.auug.org.au>
Date: Mon, 15 Sep 2014 17:01:32 +1000
Subject: [PATCH] pnfs/blocklayout: include vmalloc.h for __vmalloc
Git-commit: b262b35c2ceb989c6b89dae6b8e0c0ce466cc90d
Patch-mainline: v3.18
References: bnc#898675

Signed-off-by: Stephen Rothwell <sfr@canb.auug.org.au>
Reviewed-by: Christoph Hellwig <hch@lst.de>
Signed-off-by: Trond Myklebust <trond.myklebust@primarydata.com>
Acked-by: NeilBrown <neilb@suse.de>

---
 fs/nfs/blocklayout/extent_tree.c |    2 ++
 1 file changed, 2 insertions(+)

--- linux-3.12-SLE12.orig/fs/nfs/blocklayout/extent_tree.c
+++ linux-3.12-SLE12/fs/nfs/blocklayout/extent_tree.c
@@ -2,6 +2,8 @@
  * Copyright (c) 2014 Christoph Hellwig.
  */
 
+#include <linux/vmalloc.h>
+
 #include "blocklayout.h"
 
 #define NFSDBG_FACILITY		NFSDBG_PNFS_LD
