From: Valentina Giusti <valentina.giusti@microon.de>
Date: Mon, 4 Nov 2013 22:34:28 +0100
Patch-mainline: 3.14
Git-commit: e94acd86d48d61a5d919d807ed1efa0d8c1cd5ae
Subject: [PATCH] btrfs: replace path->slots[0] with otherwise unused
 variable 'slot'

Signed-off-by: Valentina Giusti <valentina.giusti@microon.de>
Signed-off-by: Josef Bacik <jbacik@fusionio.com>
Signed-off-by: Chris Mason <clm@fb.com>
Signed-off-by: David Sterba <dsterba@suse.cz>
---
 fs/btrfs/backref.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/fs/btrfs/backref.c
+++ b/fs/btrfs/backref.c
@@ -1691,8 +1691,8 @@ static int iterate_inode_extrefs(u64 inu
 		btrfs_set_lock_blocking_rw(eb, BTRFS_READ_LOCK);
 		btrfs_release_path(path);
 
-		item_size = btrfs_item_size_nr(eb, path->slots[0]);
-		ptr = btrfs_item_ptr_offset(eb, path->slots[0]);
+		item_size = btrfs_item_size_nr(eb, slot);
+		ptr = btrfs_item_ptr_offset(eb, slot);
 		cur_offset = 0;
 
 		while (cur_offset < item_size) {
