From: "Yan, Zheng" <zheng.z.yan@intel.com>
Date: Wed, 13 Nov 2013 14:47:19 +0800
Subject: [PATCH] ceph: use ceph_seq_cmp() to compare migrate_seq
Git-commit: d1b87809fba3e07a261080837d1ae58d790b51a6
Patch-mainline: v3.14
References: Fate#318586


Signed-off-by: Yan, Zheng <zheng.z.yan@intel.com>
Acked-by: NeilBrown <neilb@suse.com>

---
 fs/ceph/caps.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/fs/ceph/caps.c
+++ b/fs/ceph/caps.c
@@ -628,7 +628,7 @@ retry:
 	cap->cap_id = cap_id;
 	cap->issued = issued;
 	cap->implemented |= issued;
-	if (mseq > cap->mseq)
+	if (ceph_seq_cmp(mseq, cap->mseq) > 0)
 		cap->mds_wanted = wanted;
 	else
 		cap->mds_wanted |= wanted;
