From: "Yan, Zheng" <zheng.z.yan@intel.com>
Date: Sat, 1 Mar 2014 22:22:57 +0800
Subject: [PATCH] ceph: print inode number for LOOKUPINO request
Git-commit: c137a32a408af7a5635f3d0c5ddd34d270af9a3b
Patch-mainline: v3.15
References: Fate#318586


Signed-off-by: Yan, Zheng <zheng.z.yan@intel.com>
Reviewed-by: Sage Weil <sage@inktank.com>
Acked-by: NeilBrown <neilb@suse.com>

---
 fs/ceph/debugfs.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/fs/ceph/debugfs.c
+++ b/fs/ceph/debugfs.c
@@ -93,6 +93,8 @@ static int mdsc_show(struct seq_file *s,
 		} else if (req->r_path1) {
 			seq_printf(s, " #%llx/%s", req->r_ino1.ino,
 				   req->r_path1);
+		} else {
+			seq_printf(s, " #%llx", req->r_ino1.ino);
 		}
 
 		if (req->r_old_dentry) {
