From: Jianpeng Ma <jianpeng.ma@intel.com>
Date: Tue, 18 Aug 2015 10:30:38 +0800
Subject: [PATCH] ceph: cleanup use of ceph_msg_get
Git-commit: 5fdb1389e1399d6801a8c5d10952ef4153039fb2
Patch-mainline: v4.3
References: Fate#318586


Signed-off-by: Jianpeng Ma <jianpeng.ma@intel.com>
Signed-off-by: Yan, Zheng <zyan@redhat.com>
Acked-by: NeilBrown <neilb@suse.com>

---
 fs/ceph/mds_client.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

--- a/fs/ceph/mds_client.c
+++ b/fs/ceph/mds_client.c
@@ -2526,8 +2526,7 @@ out_err:
 		if (err) {
 			req->r_err = err;
 		} else {
-			req->r_reply = msg;
-			ceph_msg_get(msg);
+			req->r_reply =  ceph_msg_get(msg);
 			req->r_got_result = true;
 		}
 	} else {
