From: "Yan, Zheng" <zheng.z.yan@intel.com>
Date: Thu, 26 Jun 2014 15:25:17 +0800
Subject: [PATCH] ceph: check unsupported fallocate mode
Git-commit: 494d77bf8fd732f164b91710bd522ee38acb4c2e
Patch-mainline: v3.17
References: Fate#318586


Signed-off-by: Yan, Zheng <zheng.z.yan@intel.com>
Acked-by: NeilBrown <neilb@suse.com>

---
 fs/ceph/file.c |    3 +++
 1 file changed, 3 insertions(+)

--- a/fs/ceph/file.c
+++ b/fs/ceph/file.c
@@ -1247,6 +1247,9 @@ static long ceph_fallocate(struct file *
 	loff_t endoff = 0;
 	loff_t size;
 
+	if (mode & ~(FALLOC_FL_KEEP_SIZE | FALLOC_FL_PUNCH_HOLE))
+		return -EOPNOTSUPP;
+
 	if (!S_ISREG(inode->i_mode))
 		return -EOPNOTSUPP;
 
