From: Fengguang Wu <fengguang.wu@intel.com>
Date: Wed, 9 Apr 2014 14:11:31 +0800
Subject: [PATCH] ceph: ceph_get_parent() can be static
Git-commit: e84be11c5360df33f0ba6c39185ccba564ec7177
Patch-mainline: v3.16
References: Fate#318586


Signed-off-by: Fengguang Wu <fengguang.wu@intel.com>
Reviewed-by: Yan, Zheng <zheng.z.yan@intel.com>
Acked-by: NeilBrown <neilb@suse.com>

---
 fs/ceph/export.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/fs/ceph/export.c
+++ b/fs/ceph/export.c
@@ -169,7 +169,7 @@ static struct dentry *__get_parent(struc
 	return dentry;
 }
 
-struct dentry *ceph_get_parent(struct dentry *child)
+static struct dentry *ceph_get_parent(struct dentry *child)
 {
 	/* don't re-export snaps */
 	if (ceph_snap(child->d_inode) != CEPH_NOSNAP)
