From: "Yan, Zheng" <zheng.z.yan@intel.com>
Date: Tue, 1 Apr 2014 20:34:18 +0800
Subject: [PATCH] ceph: don't grabs open file reference for aborted request
Git-commit: 48193012873e341f08de48304e32d0499b96c60b
Patch-mainline: v3.15
References: Fate#318586


Signed-off-by: Yan, Zheng <zheng.z.yan@intel.com>
Acked-by: NeilBrown <neilb@suse.com>

---
 fs/ceph/inode.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/fs/ceph/inode.c
+++ b/fs/ceph/inode.c
@@ -1110,7 +1110,7 @@ retry_lookup:
 
 		err = fill_inode(in, &rinfo->targeti, NULL,
 				session, req->r_request_started,
-				(le32_to_cpu(rinfo->head->result) == 0) ?
+				(!req->r_aborted && rinfo->head->result == 0) ?
 				req->r_fmode : -1,
 				&req->r_caps_reservation);
 		if (err < 0) {
