From: Ilya Dryomov <idryomov@redhat.com>
Date: Mon, 2 Feb 2015 17:54:17 +0300
Subject: [PATCH] ceph: show nocephx_require_signatures and notcp_nodelay
 options
Git-commit: 2a0b61cefcd52ad63ff03aacae6d4113cdf46812
Patch-mainline: v4.0
References: Fate#318586


Signed-off-by: Ilya Dryomov <idryomov@redhat.com>
Acked-by: NeilBrown <neilb@suse.com>

---
 fs/ceph/super.c |    4 ++++
 1 file changed, 4 insertions(+)

--- a/fs/ceph/super.c
+++ b/fs/ceph/super.c
@@ -425,6 +425,10 @@ static int ceph_show_options(struct seq_
 		seq_puts(m, ",noshare");
 	if (opt->flags & CEPH_OPT_NOCRC)
 		seq_puts(m, ",nocrc");
+	if (opt->flags & CEPH_OPT_NOMSGAUTH)
+		seq_puts(m, ",nocephx_require_signatures");
+	if ((opt->flags & CEPH_OPT_TCP_NODELAY) == 0)
+		seq_puts(m, ",notcp_nodelay");
 
 	if (opt->name)
 		seq_printf(m, ",name=%s", opt->name);
