From: "Yan, Zheng" <zheng.z.yan@intel.com>
Date: Tue, 11 Feb 2014 13:23:09 +0800
Subject: [PATCH] ceph: fix ceph_removexattr()
Git-commit: 524186ace6c4dcc83975b858622a66888b018fd0
Patch-mainline: v3.14
References: Fate#318586


Signed-off-by: Yan, Zheng <zheng.z.yan@intel.com>
Acked-by: NeilBrown <neilb@suse.com>

---
 fs/ceph/xattr.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/fs/ceph/xattr.c
+++ b/fs/ceph/xattr.c
@@ -453,7 +453,7 @@ static int __remove_xattr(struct ceph_in
 			  struct ceph_inode_xattr *xattr)
 {
 	if (!xattr)
-		return -EOPNOTSUPP;
+		return -ENODATA;
 
 	rb_erase(&xattr->node, &ci->i_xattrs.index);
 
