From: "Yan, Zheng" <zyan@redhat.com>
Date: Fri, 16 Jan 2015 10:54:43 +0800
Subject: [PATCH] client: include kernel version in client metadata
Git-commit: a6a5ce4f0df9146ba8cb61121b80aa191fbb1f04
Patch-mainline: v4.0
References: Fate#318586


Signed-off-by: Yan, Zheng <zyan@redhat.com>
Acked-by: NeilBrown <neilb@suse.com>

---
 fs/ceph/mds_client.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/fs/ceph/mds_client.c
+++ b/fs/ceph/mds_client.c
@@ -845,8 +845,9 @@ static struct ceph_msg *create_session_o
 	struct ceph_options *opt = mdsc->fsc->client->options;
 	void *p;
 
-	const char* metadata[3][2] = {
+	const char* metadata[][2] = {
 		{"hostname", utsname()->nodename},
+		{"kernel_version", utsname()->release},
 		{"entity_id", opt->name ? opt->name : ""},
 		{NULL, NULL}
 	};
