From: "Yan, Zheng" <zyan@redhat.com>
Date: Mon, 17 Nov 2014 10:01:03 +0800
Subject: [PATCH] ceph: support inline data feature
Git-commit: 65a22662bfe1a84d72b9bbd9146b6782b9e53478
Patch-mainline: v3.19
References: Fate#318586


Signed-off-by: Yan, Zheng <zyan@redhat.com>
Acked-by: NeilBrown <neilb@suse.com>

---
 fs/ceph/super.c |    5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

--- a/fs/ceph/super.c
+++ b/fs/ceph/super.c
@@ -513,9 +513,10 @@ static struct ceph_fs_client *create_fs_
 					struct ceph_options *opt)
 {
 	struct ceph_fs_client *fsc;
-	const unsigned supported_features =
+	const u64 supported_features =
 		CEPH_FEATURE_FLOCK |
-		CEPH_FEATURE_DIRLAYOUTHASH;
+		CEPH_FEATURE_DIRLAYOUTHASH |
+		CEPH_FEATURE_MDS_INLINE_DATA;
 	const unsigned required_features = 0;
 	int page_count;
 	size_t size;
