From: "Yan, Zheng" <zyan@redhat.com>
Date: Mon, 27 Apr 2015 11:02:35 +0800
Subject: [PATCH] libceph: properly release STAT request's raw_data_in
Git-commit: 66ba609f7b96096116ca7bbc21ec6922ea41a992
Patch-mainline: v4.2
References: Fate#318586


Signed-off-by: Yan, Zheng <zyan@redhat.com>
Reviewed-by: Alex Elder <elder@linaro.org>
Acked-by: NeilBrown <neilb@suse.com>

---
 net/ceph/osd_client.c |    3 +++
 1 file changed, 3 insertions(+)

--- a/net/ceph/osd_client.c
+++ b/net/ceph/osd_client.c
@@ -447,6 +447,9 @@ static void osd_req_op_data_release(stru
 		ceph_osd_data_release(&op->xattr.request_data);
 		ceph_osd_data_release(&op->xattr.response_data);
 		break;
+	case CEPH_OSD_OP_STAT:
+		ceph_osd_data_release(&op->raw_data_in);
+		break;
 	default:
 		break;
 	}
