From: "Yan, Zheng" <zheng.z.yan@intel.com>
Date: Mon, 21 Jul 2014 10:15:48 +0800
Subject: [PATCH] ceph: check zero length in ceph_sync_read()
Git-commit: d0d0db2268cc343c2361c83510d8e9711021fcce
Patch-mainline: v3.17
References: Fate#318586


Signed-off-by: Yan, Zheng <zheng.z.yan@intel.com>
Acked-by: NeilBrown <neilb@suse.com>

---
 fs/ceph/file.c |    3 +++
 1 file changed, 3 insertions(+)

--- a/fs/ceph/file.c
+++ b/fs/ceph/file.c
@@ -454,6 +454,9 @@ static ssize_t ceph_sync_read(struct kio
 	dout("sync_read on file %p %llu~%u %s\n", file, off,
 	     (unsigned)len,
 	     o_direct ? "O_DIRECT" : "");
+
+	if (!len)
+		return 0;
 	/*
 	 * flush any page cache pages in this range.  this
 	 * will make concurrent normal and sync io slow,
