From: "Yan, Zheng" <zheng.z.yan@intel.com>
Date: Tue, 1 Apr 2014 20:34:56 +0800
Subject: [PATCH] ceph: flush cap release queue when trimming session caps
Git-commit: a56371d9d920799ebb88c196aa018e76fc46554f
Patch-mainline: v3.15
References: Fate#318586


Signed-off-by: Yan, Zheng <zheng.z.yan@intel.com>
Acked-by: NeilBrown <neilb@suse.com>

---
 fs/ceph/mds_client.c |    3 +++
 1 file changed, 3 insertions(+)

--- a/fs/ceph/mds_client.c
+++ b/fs/ceph/mds_client.c
@@ -1311,6 +1311,9 @@ static int trim_caps(struct ceph_mds_cli
 			trim_caps - session->s_trim_caps);
 		session->s_trim_caps = 0;
 	}
+
+	ceph_add_cap_releases(mdsc, session);
+	ceph_send_cap_releases(mdsc, session);
 	return 0;
 }
 
