From 110e558f132d3665356c31c21740efc397e35357 Mon Sep 17 00:00:00 2001
From: Mel Gorman <mgorman@suse.de>
Date: Mon, 2 Jun 2014 14:51:31 +0100
Subject: [PATCH] mm: filemap: Avoid unnecessary barriers and waitqueue lookups
 -fix

References: VM/FS Performance
Patch-mainline: No (Expected 3.17)

Signed-off-by: Mel Gorman <mgorman@suse.de>
---
 mm/swap.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/mm/swap.c b/mm/swap.c
index b030567..6e6549b 100644
--- a/mm/swap.c
+++ b/mm/swap.c
@@ -78,6 +78,7 @@ static void __put_compound_page(struct page *page)
 {
 	compound_page_dtor *dtor;
 
+	__ClearPageWaiters(page);
 	__page_cache_release(page);
 	dtor = get_compound_page_dtor(page);
 	(*dtor)(page);
