From: jbeulich@suse.com
Subject: fix kABI after "x86: use custom dma_get_required_mask()"
Patch-mainline: Never, SUSE kABI

--- a/arch/x86/kernel/pci-dma-xen.c
+++ b/arch/x86/kernel/pci-dma-xen.c
@@ -270,7 +270,9 @@ u64 dma_get_required_mask(struct device 
 
 	return DMA_BIT_MASK(__fls(max_mfn - 1) + 1 + PAGE_SHIFT);
 }
+#ifndef __GENKSYMS__
 EXPORT_SYMBOL_GPL(dma_get_required_mask);
+#endif
 
 static int check_pages_physically_contiguous(unsigned long pfn,
 					     unsigned int offset,
--- a/drivers/base/platform.c
+++ b/drivers/base/platform.c
@@ -936,6 +936,9 @@ u64 dma_get_required_mask(struct device 
 	}
 	return mask;
 }
+#endif
+#if !defined(ARCH_HAS_DMA_GET_REQUIRED_MASK) || \
+    (defined(__GENKSYMS__) && defined(CONFIG_X86) && defined(CONFIG_XEN))
 EXPORT_SYMBOL_GPL(dma_get_required_mask);
 #endif
 
