From: Takashi Iwai <tiwai@suse.de>
Subject: Fix kABI breakage of dccp in 4.4.56
Patch-mainline: Never, openSUSE-42.2 only
References: stable-4.4.56

Signed-off-by: Takashi Iwai <tiwai@suse.de>

---
 include/linux/dccp.h |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/include/linux/dccp.h
+++ b/include/linux/dccp.h
@@ -163,10 +163,12 @@ struct dccp_request_sock {
 	__u64			 dreq_isr;
 	__u64			 dreq_gsr;
 	__be32			 dreq_service;
-	spinlock_t		 dreq_lock;
 	struct list_head	 dreq_featneg;
 	__u32			 dreq_timestamp_echo;
 	__u32			 dreq_timestamp_time;
+#ifndef __GENKSYMS__
+	spinlock_t		 dreq_lock;
+#endif
 };
 
 static inline struct dccp_request_sock *dccp_rsk(const struct request_sock *req)
