From: jbeulich@suse.com
Subject: fix build with gcc 6
Patch-mainline: Never, SUSE-Xen specific
References: none

--- a/drivers/acpi/processor_idle.c
+++ b/drivers/acpi/processor_idle.c
@@ -1055,7 +1055,6 @@ static int acpi_processor_setup_cpuidle_
 }
 #else
 static void acpi_processor_setup_cpuidle_cx(struct acpi_processor *pr) {}
-static void acpi_processor_setup_cpuidle_states(struct acpi_processor *pr) {}
 #endif /* CONFIG_PROCESSOR_EXTERNAL_CONTROL */
 
 int acpi_processor_hotplug(struct acpi_processor *pr)
@@ -1099,8 +1098,6 @@ int acpi_processor_hotplug(struct acpi_p
 
 int acpi_processor_cst_has_changed(struct acpi_processor *pr)
 {
-	int cpu;
-	struct acpi_processor *_pr;
 	struct cpuidle_device *dev;
 
 	if (disabled_by_idle_boot_param())
@@ -1115,6 +1112,7 @@ int acpi_processor_cst_has_changed(struc
 	if (!pr->flags.power_setup_done)
 		return -ENODEV;
 
+#ifndef CONFIG_PROCESSOR_EXTERNAL_CONTROL
 	/*
 	 * FIXME:  Design the ACPI notification to make it once per
 	 * system instead of once per-cpu.  This condition is a hack
@@ -1122,6 +1120,8 @@ int acpi_processor_cst_has_changed(struc
 	 */
 
 	if (pr->id == 0 && cpuidle_get_driver() == &acpi_idle_driver) {
+		int cpu;
+		struct acpi_processor *_pr;
 
 		/* Protect against cpu-hotplug */
 		get_online_cpus();
@@ -1155,6 +1155,7 @@ int acpi_processor_cst_has_changed(struc
 		cpuidle_resume_and_unlock();
 		put_online_cpus();
 	}
+#endif
 
 	return 0;
 }
