From: jbeulich@suse.com
Subject: blkfront: fix an error path memory leak
Patch-mainline: Never, SUSE-Xen specific
References: luckily none so far

This was overlooked during the original forward porting process.

--- a/drivers/xen/blkback/xenbus.c
+++ b/drivers/xen/blkback/xenbus.c
@@ -361,8 +361,11 @@ static void backend_changed(struct xenbu
 	}
 
 	/* Front end dir is a number, which is used as the handle. */
-	if (kstrtoul(strrchr(dev->otherend, '/') + 1, 0, &handle))
+	if (kstrtoul(strrchr(dev->otherend, '/') + 1, 0, &handle)) {
+		kfree(be->mode);
+		be->mode = NULL;
 		return;
+	}
 
 	be->major = major;
 	be->minor = minor;
