From: jbeulich@suse.com
Subject: fix build with gcc 6
Patch-mainline: Never, SUSE-Xen specific
References: none

--- a/drivers/acpi/processor_idle.c
+++ b/drivers/acpi/processor_idle.c
@@ -1099,8 +1099,6 @@ int acpi_processor_hotplug(struct acpi_p
 
 int acpi_processor_cst_has_changed(struct acpi_processor *pr)
 {
-	struct cpuidle_device *dev;
-
 	if (disabled_by_idle_boot_param())
 		return 0;
 
@@ -1123,6 +1121,7 @@ int acpi_processor_cst_has_changed(struc
 	if (pr->id == 0 && cpuidle_get_driver() == &acpi_idle_driver) {
 		int cpu;
 		struct acpi_processor *_pr;
+		struct cpuidle_device *dev;
 
 		/* Protect against cpu-hotplug */
 		get_online_cpus();
